amazon/ionbenchmark/ion_benchmark_cli.py

```diff
@@ -264,33 +431,126 @@ def write_micro_benchmark_simpleion(iterations, warmups, c_extension, obj, file,
     return file_size, result_with_gc
 
 
+# Benchmarks JSON/CBOR APIs
+def write_micro_benchmark(iterations, warmups, c_extension, file, binary, memory_profiling, format_option, io_type):
+    file_size = Path(file).stat().st_size / BYTES_TO_MB
+    obj = generate_json_and_cbor_obj_for_write(file, format_option)
+    # GC refers to reference cycles, not reference count
+    setup_with_gc = generate_setup(gc=False)
+
+    test_func = generate_write_test_code(obj, memory_profiling=memory_profiling, format_option=format_option,
+                                         io_type=io_type, binary=binary)
+
+    # warm up
+    timeit.timeit(stmt=test_func, setup=setup_with_gc, number=warmups)
+
+    # iteration
+    result_with_gc = timeit.timeit(stmt=test_func, setup=setup_with_gc, number=iterations) / iterations
+
+    return file_size, result_with_gc
+
+
+# Generates benchmark code for json dump API
+def generate_write_test_code(obj, memory_profiling, format_option, io_type, binary):
```

Comment:
Is there a reason Ion can't follow this path?

Reply:
same as above comment.